### DISCUSSION TITLE
Resuming training

### DISCUSSION THREAD

**Discussion by infamix** [erfanzar/EasyDeL#68] [Q&A] [answered]
If I resume training from a ckpt the dataset will continue right where it left off, right?

**Comment by erfanzar** [comment, marked as answer]
No it won't start where it left off from dataset but if the data is saved by easydel savers itself optimizers will resume on previous ckpt and load from last step

**Reply by infamix** [reply]
Is this all that's needed to resume training? 
```
trainer = CausalLMTrainer(
    train_args,
    dataset_train,
    ckpt_path="/kaggle/input/easydel-sq-6000/EasyDeL/lib/python/easydel_ckpt/EasyDelLLama2/EasyDelLLama2-1.5074825286865234-6000"
)
```
Because it didn't happen and I'm a little confused.

**Reply by erfanzar** [reply]
Yes that all you need and then you call .train function to train model

**Reply by infamix** [reply]
I understand but it didn't work as intended, instead the training started from 0.

**Reply by erfanzar** [reply]
Yes it will start from step 0 but resume the previous process

**Reply by infamix** [reply]
Negative, the loss and the optimizer graphs (etc.) looked exactly like you'd start from 0 and not like you'd resume from a checkpoint.
